opaque_keys/edx/keys.py

```diff
@@ -89,6 +89,36 @@ def make_asset_key(self, asset_type: str, path: str) -> AssetKey:  # pragma: no
         raise NotImplementedError()
 
 
+class LibCollectionKey(LearningContextKey):
```

Comment:
```suggestion
class LibraryCollectionKey(OpaqueKey):
```

1. My sense is that a collection is not a learning context. More concretely, I can't imagine us wanting to store learner state against a collection, the way that we store learner state against a CourseKey or (for previewing purposes) a LibraryKey. Would it cause any issue to simply subclass OpaqueKey directly?
2. For the sake of the API, let's be consistent between "Lib" and "Library".  We have `lib:` in the keys, so I like that you chose `lib-collection:`. But we spell out `LibraryLocatorV2`, so I'd prefer if you spelled out `LibraryCollectionKey` and `LibraryCollectionLocator` as well.

Reply:
Agreed that a collection is not a learning context and it should not subclass `LearningContextKey`. However, it can have a `library_key ` _field_ if that makes sense, since it does related to a library.

Reply:
@bradenmacdonald Sure, having a library_key field makes sense.

@ChrisChV Is there any reason we need both a field `LibraryCollectionLocator.lib_key` and an identical property `LibraryCollectionLocator.library_key` ? Could there just be a field named `LibraryCollectionLocator.library_key` ? Or does that play poorly with inheritance?

Reply:
I combined both fields here: https://github.com/openedx/opaque-keys/pull/335/commits/ce740812b6874493989bdd1c6502378911c51e91